--- norarefy-source/minimal-rarefy-example/minimal-rarefy-example.Rmd	2022-11-23 16:00:57
+++ patches/minimal-rarefy-example.Rmd	2022-11-28 18:27:08
@@ -1,4 +1,6 @@
-`r opts_chunk$set(cache=TRUE, fig.width=10)`
+`r opts_chunk$set(cache=TRUE, fig.width=10)
+opts_chunk$set(fig.path="norarefy-source/minimal-rarefy-example/")
+`
 <link href="http://joey711.github.com/phyloseq/markdown.css" rel="stylesheet"></link>
 
 
